source/elements/oneTBB/source/named_requirements/algorithms/par_for_each_body.rst

```diff
@@ -7,54 +7,45 @@ ParallelForEachBody
-
-.. note::
+    Process the received item.
 
-    ``ItemType`` may be optionally passed to ``Body::operator()`` by reference.
-    ``const`` and ``volatile`` type qualifiers are also applicable.
+----------------------------------------------------------------
 
-Terms
------
+Variant 2:
 
-* ``iterator`` determines the type of the iterator passed into the ``parallel_for_each`` algorithm,
-  which is ``decltype(std::begin(c))`` for the overloads that accept the ``Container`` template argument or ``InputIterator``.
-* ``value_type`` - the type ``std::iterator_traits<iterator>::value_type``.
-* ``reference`` -  the type ``std::iterator_traits<iterator>::reference``.
+.. cpp:function:: void Body::operator()( ReferenceType item, oneapi::tbb::feeder<ItemType>& feeder ) const
+                  void Body::operator()( ItemType&& item, oneapi::tbb::feeder<ItemType>& feeder ) const
 
-``oneapi::tbb::parallel_for_each`` requires the ``Body::operator()`` call with an object of the ``reference`` type to be well-formed if
-the ``iterator`` meets the `Forward iterator` requirements described in the [forward.iterators] section of the 
-ISO C++ Standard.
+    Process the received item. May invoke the ``feeder.add`` function to spawn additional items.
+    The ``Body::operator()`` must accept both ``ReferenceType`` and ``ItemType&&`` values as the first argument.
 
-`oneapi::tbb::parallel_for_each algorithm <../../algorithms/functions/parallel_for_each_func>`_
-requires the ``Body::operator()`` call with an object of type ``const value_type&`` or ``value_type&&`` to be well-formed if following requirements are met:
+-----------------------------------------------------------------
 
-* the iterator meets the `Input iterator` requirements described in the [input.iterators] section of the ISO C++ Standard
-* the iterator does not meet the `Forward iterator` requirements described in the [forward.iterators] section of the ISO C++ Standard
+where
 
-.. caution::
+* ``ItemType`` is ``std::iterator_traits<Iterator>::value_type`` for the type of the iterator
+  the ``parallel_for_each`` algorithm operates with, and
+* ``ReferenceType`` is ``std::iterator_traits<Iterator>::reference`` if the iterator type is
+  a `forward iterator` as described in the [forward.iterators] section of the ISO C++ Standard,
+* otherwise, ``ReferenceType`` is ``ItemType&&``.
 
-  If the ``Body`` only takes non-const lvalue reference to the ``value_type``, the requirements described above
-  are violated, and the program can be ill-formed.
+.. note::
 
-Additional elements submitted into ``oneapi::tbb::parallel_for_each`` through the ``feeder::add`` are passed to the ``Body`` as rvalues. In this case, the corresponding
-execution of the ``Body`` is required to be well-formed.
+    The usual rules for :ref:`pseudo-signatures <pseudo_signatures>` apply.
+    Therefore, ``Body::operator()`` may optionally take items by value.
+    ``const`` and ``volatile`` type qualifiers are also applicable.
```

Comment:
Do we really want to mention volatile? C++ standard tries to avoid it these days until absolutely necessary (based on my understanding, of course). Why should we?